src/model/guild/mod.rs

```diff
@@ -1616,19 +1616,10 @@ impl Guild {
 
     /// Gets a list of all the members (satisfying the status provided to the function) in this
     /// guild.
-    #[must_use]
-    pub fn members_with_status(&self, status: OnlineStatus) -> Vec<&Member> {
-        let mut members = vec![];
-
-        for (&id, member) in &self.members {
-            if let Some(presence) = self.presences.get(&id) {
-                if status == presence.status {
-                    members.push(member);
-                }
-            }
-        }
-
-        members
+    pub fn members_with_status(&self, status: OnlineStatus) -> impl Iterator<Item = &Member> {
```

Comment:
Why change the return type on this? I think probably most people would expect a `Vec`, so having to explicitly collect into one is an ergonomics hit.

Reply:
If they want a vector, they can `.collect()` pretty easily. This is just more flexible for performance reasons. It's the same as `CacheRef` where we take an easily fixable ergonomics hit to be more flexible for performance.